src/main/java/org/folio/service/request/RequestBatchResourceService.java

```diff
@@ -0,0 +1,91 @@
+package org.folio.service.request;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.folio.rest.jaxrs.model.Request;
+import org.folio.service.BatchResourceService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rits.cloning.Cloner;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+public class RequestBatchResourceService {
+  private static final Logger LOG = LoggerFactory.getLogger(RequestBatchResourceService.class);
+
+  private final BatchResourceService batchResourceService;
+  private final Cloner cloner;
+
+
+  public RequestBatchResourceService(BatchResourceService batchResourceService) {
+    this.batchResourceService = batchResourceService;
+    this.cloner = new Cloner();
+  }
+
+  /**
+   * This method executes batch update for the request table.
+   * The batch package must include request copies with removed positions
+   * as a first part of transaction in order to pass 'itemId - position' constraint.
+   * <p>
+   * Example:
+   * Let's say we have following requests in the batch:
+   * 1. Request A.
+   * 2. Request B.
+   * 3. Request C.
+   * <p>
+   * Then actual batch package, that will be executed, is following:
+   * 1. Copy of Request A with position = null.
+   * 1. Copy of Request B with position = null.
+   * 1. Copy of Request C with position = null.
+   * 4. Request A.
+   * 5. Request B.
+   * 6. Request C.
+   *
+   * @param requests        - List of requests to execute in batch.
+   * @param onFinishHandler - Callback function.
+   */
+  public void executeRequestBatchUpdate(
+    List<Request> requests, Handler<AsyncResult<Void>> onFinishHandler) {
+
+    LOG.debug("Removing positions for all request to go through positions constraint");
+    // We have to include requests with positions removed to the batch package
+    // to go through itemId-position unique constraint.
+    List<Request> requestsWithRemovedPositions = removePositionsForRequests(requests);
+
+    List<Request> allRequestsToUpdate = new ArrayList<>(requestsWithRemovedPositions);
+    allRequestsToUpdate.addAll(requests);
+
+    LOG.info("Executing batch update, total records to update [{}] (including remove positions)",
+      allRequestsToUpdate.size()
+    );
+
+    batchResourceService.executeBatchUpdate(allRequestsToUpdate,
```

Comment:
Instead of doing this cloning with `null` positions, can we just pass in some pre-batch SQL that removes the position for each request ID, like this code:
https://github.com/folio-org/mod-circulation-storage/blob/fa913ecea1fdd9af14dc69063b5c1e533db2b9a4/src/main/java/org/folio/support/ExpirationTool.java#L235-L239

It think pre/post-batch SQL might be good options for the batch executor in order to do other stuff within the batch transaction.

Reply:
I prefer the SQL based approach over the deep copy and an update statement per record.